lib/parallel.js

```diff
@@ -228,7 +242,10 @@ function createParallelRouter(spec) {
 
         const currentState = machine.machine.current();
 
-        if (currentSectionId === currentState.context.routes.initial) {
+        if (
+            currentState.context.routes.referrer &&
+            currentSectionId === currentState.context.routes.initial
+        ) {
```

Comment:
At the moment, the validator requires a referrer to pass, so I don't think this could ever be necessary in practice?